astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -17,12 +17,56 @@
 from ... import _erfa as erfa
 
 from .icrs import ICRS
+from .itrs import ITRS
 from .gcrs import GCRS
 from .cirs import CIRS
 from .hcrs import HCRS
 from .utils import get_jd12
 
 
+# helper function to find GCRS position and velocity of observatory
+def get_gcrs_posvel(time, frame):
+    """
+    Given frame with ```obsgeoloc``` and ```obsgeovel```, find GCRS position and velocity.
+
+    ```obsgeoloc``` and ```obsgeovel``` are specified in the ITRS coordinate frame.
+    This function converts the position and velocity into the GCRS coordinate frame.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        The time at which to calculate GCRS position and velocity.
+    frame : `~astropy.coordinates.BaseCoordinateFrame
+        The frame to use. Must have ```obsgeoloc``` and ```obsgeovel``` attributes.
+
+    Returns
+    --------
+    pv : np.ndarray
+        GCRS position and velocity in m and m/s
+    """
+    try:
+        obsgeoloc = ITRS(frame.obsgeoloc, obstime=time)
+        obsgeovel = ITRS(frame.obsgeovel, obstime=time)
+    except AttributeError:
+        raise ValueError("frame must have obsgeoloc and obsgeovel attributes set")
+
+    # transform to GCRS. Just a rotation matrix so velocity should be fine?
+    geocentric_frame = GCRS(obstime=time)
+    pos_arr = obsgeoloc.transform_to(geocentric_frame).cartesian.xyz.to(u.m).value
+    vel_arr = obsgeovel.transform_to(geocentric_frame).cartesian.xyz.to(u.m/u.s).value
+
+    # now add rotational velocity of GCRS frame w.r.t to ITRS
+    # rotation of earth
+    V = u.Quantity([0, 0, 7.292115855306589e-5])*u.rad/u.s
```

Comment:
Can this be defined as a constant, along with a reference for where it comes from?